### ISSUE TITLE
Dependency Conflict While Installing Plaquette Using Pip

### ISSUE THREAD

**Issue by ishdhand** [qc-design/plaquette#2]
Description:
While attempting to install the Plaquette package (version 0.0.1a0) using pip, dependency conflicts arose due to incompatible versions of numpy being required by different packages.

Reproduction Steps:

- Use Python version 3.10.6 on Apple silicon mac.
- Attempt to install Plaquette with the `pip install plaquette`.

Result:
After successful uninstallation of numpy 1.23.5 and installation of numpy 1.24.3, the pip dependency resolver reported the following conflicts:
 
- The Galois package (version 0.3.3) requires numpy version <1.24,>=1.21.0, but the installed version of numpy is 1.24.3.
- The Numba package (version 0.56.4) requires numpy version <1.24,>=1.18, but the installed version of numpy is 1.24.3.

Error message:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
galois 0.3.3 requires numpy<1.24,>=1.21.0, but you have numpy 1.24.3 which is incompatible.
numba 0.56.4 requires numpy<1.24,>=1.18, but you have numpy 1.24.3 which is incompatible.
```

**Comment by massarom** [comment]
Can you please try to install this in a clean/new virtual environment?

Those packages that `pip` is complaining about are not dependencies of `plaquette`. You might be trying to install plaquette into a busy environment, so to say, where some previously installed packages would be broken by the actual requirements of `plaquette`.

This also explains why, error notwithstanding, everything works because you have installed new dependencies that are incompatible with old packages currently installed in your environment, but plaquette does not use these now-broken packages.

TL;DR in a new environment I expect this error to never appear.
